ligo/src/types.mligo

```diff
@@ -21,6 +21,26 @@ type ledger_key = address * token_id
 type ledger_value = nat
 type ledger = (ledger_key, ledger_value) big_map
 
+type freeze_history_entry =
+  { period : nat
+  ; staked : nat
+  ; unstaked : nat
+  }
+
+type freeze_history = (address, freeze_history_entry list) big_map
+
+// Utility type to facilitate manipulation between internal functions
+// Note: it only has one "staked" as only past tokens can be staked
+type compact_freeze_history =
+  { current_period : nat
+  ; staked : nat
+  ; current_unstaked : nat
+  ; past_unstaked : nat
+  }
+
+// The count of actually staked frozen tokens.
+type staked = (address, nat) big_map
```

Comment:
I think this is a left-over that's no longer needed.

Reply:
Removed